shell.nix

```diff
@@ -4,15 +4,29 @@
 }:
 
 let
-  iohk-ops = (import ./default.nix {}).iohk-ops;
-  ioAlias  = if io-dev-mode
-             then ''runhaskell -iiohk iohk/iohk-ops.hs''
-             else ''${iohk-ops}/bin/iohk-ops'';
-in pkgs.lib.overrideDerivation iohk-ops.env
-   (old: {
-     shellHook = ''
-       function io {
-        ${ioAlias} "$@"
-       }
-     '';
-    })
+  iohkpkgs = import ./default.nix {};
+  iohk-ops = iohkpkgs.iohk-ops;
+  justIo = pkgs.runCommand "shell" {
+    buildInputs = [ iohk-ops ];
+    passthru = {
+      inherit ioSelfBuild withAuxx;
+    };
+  } "echo use nix-shell";
+  ioSelfBuild = pkgs.lib.overrideDerivation iohk-ops.env (drv: {
+    shellHook = ''
+      function io {
+        runhaskell -iiohk iohk/iohk-ops.hs
+      }
+      function ghcid-io {
+        ${pkgs.haskellPackages.ghcid}/bin/ghcid -c "ghci -iiohk iohk/iohk-ops.hs"
+      }
+    '';
+  });
+  withAuxx = pkgs.runCommand "shell" {
```

Comment:
@cleverca22, so this is to be used like:
```
nix-shell -A withAuxx
```
right?

Reply:
yep